docs/markdown/snippets/option-array-type.md

```diff
@@ -0,0 +1,17 @@
+# An array type for user options
+
+Previously to have an option that took more than one value a string value would
+have to be created and split, but validating this was difficult. A new array type
+has been added to the meson_options.txt for this case. It works like a 'combo', but
+allows more than one option to be passed. When used on the command line (with -D),
+values are passed as a comma separated list.
+
+    ```meson
+    option('array_opt', type : 'array', choices : ['one', 'two', 'three'], value : 'one')
```

Comment:
Maybe we should require `value:` to always be an array type? In this case that would then be `value : ['one']`. Otherwise people will constantly think that `value:` can be a comma-separated list when they see a single-value `value` kwarg.